packages/server/src/sdk/app/ai/helpers/table.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  DocumentType,
+  FieldType,
+  GenerateTablesResponse,
+  SourceName,
+  TableSchema,
+  TableSourceType,
+} from "@budibase/types"
+import sdk from "../../.."
+import { context, utils } from "@budibase/backend-core"
+
+export async function generateTables(
+  tables: { name: string; primaryDisplay: string; schema: TableSchema }[]
+) {
+  const count = (await sdk.datasources.fetch()).length
+  const { id: dsId } = await context.getAppDB().put({
+    _id: `${DocumentType.DATASOURCE}_bb_internal_${utils.newid()}`,
+    name: `Test ${count}`,
+    type: "budibase",
+
+    source: SourceName.BUDIBASE,
+    config: {},
+  })
+
+  const createdTables: GenerateTablesResponse["createdTables"] = []
+  const tableIds: Record<string, string> = {}
+
+  for (const table of tables) {
+    const createdTable = await sdk.tables.create({
+      ...table,
+      sourceId: dsId,
+      schema: {},
+      primaryDisplay: undefined,
+      sourceType: TableSourceType.INTERNAL,
+      type: "table",
+    })
+
+    createdTables.push({ id: createdTable._id!, name: createdTable.name })
+    tableIds[table.name] = createdTable._id!
+  }
+
+  for (const table of tables) {
+    for (const field of Object.values(table.schema)) {
+      if (field.type === FieldType.LINK) {
+        const linkedTable = createdTables.find(t => t.name === field.tableId)
+        if (!linkedTable) {
+          throw `Table ${field.tableId} not found in the json response.`
```

Comment:
If this happens, are we leaving the user with some half-created tables? Should we clean them up?